### PR TITLE
Add Geos

### PR DESCRIPTION
**Pull request by KGB99**

### CODE REVIEW

**Comment by KGB99** [comment]
This seems to be a longer existing problem in emscripten:
https://github.com/emscripten-core/emscripten/issues/22005
https://github.com/emscripten-core/emscripten/pull/11087
https://github.com/Keno/julia-wasm/issues/27

If cmake is executed with `-DBUILD_TESTING=OFF` the build seems to succeed, however im not sure how wise it is to ignore all tests.

**Comment by wolfv** [comment]
Maybe you can find another way to test this? Or remove tests that make the build fail?

**Comment by KGB99** [comment]
I started working on a patch where i just remove the tests that fail. Should be working soon hopefully :)

**Comment by wolfv** [comment]
Looks good to me although I couldn't really see if the tests are executed or not.

**Comment by KGB99** [comment]
The tests indeed did not run before. I tried a lot of things in the recipe such as:
```
tests:
  - script:
      - node bin/test_geos_unit.js
    requirements:
      build:
        - nodejs
    files:
      source:
        - tests/
        - bin/
        - bin/test_geos_unit.js 
```
However I constantly get file not found errors regarding `bin/test_geos_unit.js`. I think there is something im not understanding right.
I will add some existence tests for now.

**Comment by wolfv** [comment]
Hey @KGB99 - that's because the tests are not installed into the final package. You would have to execute them as part of your build script. 

You might be able to invoke them with CMake / `ctest`.

File: recipes/recipes_emscripten/geos/build.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+emmake make -j${CPU_COUNT} #${VERBOSE_CM}
+
+if [[ "${OSX_ARCH}" = "x86_64" ]]; then
+    # See https://github.com/libgeos/geos/issues/930
+    CTEST_EXCLUDE=unit-geom-Envelope
+fi
+
+if [[ "${CONDA_BUILD_CROSS_COMPILATION}" != "1" ]]; then
+    ctest --output-on-failure --exclude-regex ${CTEST_EXCLUDE}
```
Comment (wolfv) [inline recipes/recipes_emscripten/geos/build.sh]: If you enable this line, do the tests run? I.e. remove the `if`.

Comment (KGB99) [reply recipes/recipes_emscripten/geos/build.sh]: yes they do! thanks

**Review by IsabelParedes — APPROVED** [review]

**Pull request merged**